common-libs/common/src/shared/seedTools.ts

```diff
@@ -10,35 +10,46 @@ export type KeyOptions = {
   keyTypes: KeyAlgorithmType[]
 }
 
+export type ParseDecryptedSeedResult = {
+  seed: Buffer
+  didMethod: string
+  seedHexWithMethod: string
+  fullSeedHex: string
+  externalKeys?: any[]
+  metadata?: Record<string, any>
+}
+
 export const getDecryptedSeedParts = (decryptedSeed: string) => {
   const seedParts = decryptedSeed.split('++')
   const seedHex = seedParts[0]
   const didMethod = seedParts[1]
   const base64EncodedKeys = seedParts[2]
+  const base64EncodedMetadata = seedParts[3]
```

Comment:
Now I see there was a bug in original implementation, and this bug continues into this PR as well.

Unfortunately, since keys are base64-encoded, they can contain `++` substrings (as opposed to `seedHex` and `didMethod`).
And if we just split `seedParts` by `++`, we'll end up splitting these keys.

We should use something else as a separator beyond didMethod, and probably also do splitting manually.

Reply:
Also I would suggest to put an end to this expansion of decryptedSeed format, and do something like:
```
`${seedBinary}` // extremely old format
`${seedHex}++${didMethod}` // old format
`${seedHex}++${didMethod}++${base64EncodedKeys, can contain '++'}` // old format, only appeared in SDK v6 beta, so not used much
`${seedHex}++${didMethod};additionalData=${base64EncodedData}` // final format, contains both keys and metadata and all the other data we might want to add in the future
```

Reply:
`${seedHex}++${didMethod}++;additionalData:${base64EncodedData}`
so, base64EncodedData this also should contains base64EncodedKeys as well with meta?

Reply:
Yes, base64EncodedData can contain JSON with `keys` and `metadata`.

Of course this will make seed slightly larger (because we'll base64 encode keys that are already base64-encoded).

Or maybe if that is a concern we could use BSON instead of JSON.

Reply:
It could be an option, let's try this format now